packages/cozy-harvest-lib/src/components/KonnectorModal.jsx

```diff
@@ -78,18 +73,73 @@ export class KonnectorModal extends PureComponent {
     }, 50)
   }
 
+  componentDidUpdate(prevProps) {
+    if (this.props.accountId && this.props.accountId !== prevProps.accountId) {
+      this.loadSelectedAccountId()
+    }
+  }
+
+  requestAccountChange(account, trigger) {
+    // This component can either defer the account switching to a parent component through the onAccountChange prop, or handle the change itself if the prop is missing
+    const { onAccountChange } = this.props
+    return onAccountChange
+      ? onAccountChange(account)
+      : this.fetchAccount(trigger)
+  }
+
+  loadSelectedAccountId() {
+    const selectedAccountId = this.props.accountId
+    const { accounts } = this.state
+
+    const matchingTrigger = get(
+      accounts.find(account => account.account._id === selectedAccountId),
+      'trigger'
+    )
+
+    if (matchingTrigger) this.fetchAccount(matchingTrigger)
+  }
+
+  requestAccountCreation() {
+    const { createAction } = this.props
+    if (createAction) createAction()
+    else this.setState({ addingAccount: true })
+  }
+
+  async endAccountCreation(trigger) {
+    this.setState({ addingAccount: false })
+    const account = await this.fetchAccount(trigger)
+
+    if (account) {
+      this.setState(prevState => ({
+        ...prevState,
+        accounts: [
+          ...prevState.accounts,
+          {
+            account,
+            trigger
+          }
+        ]
+      }))
+    }
+  }
+
   async fetchAccounts() {
     const triggers = this.props.konnector.triggers.data
     const { findAccount } = this.props
-    const accounts = await Promise.all(
-      triggers.map(async trigger => {
-        return {
-          account: await findAccount(triggersModel.getAccountId(trigger)),
-          trigger
-        }
-      })
-    )
-    this.setState({ accounts })
+    this.setState({ fetchingAccounts: true })
+    try {
+      const accounts = (await Promise.all(
+        triggers.map(async trigger => {
+          return {
+            account: await findAccount(triggersModel.getAccountId(trigger)),
+            trigger
+          }
+        })
+      )).filter(({ account }) => !!account)
```

Comment:
why this filter?

Reply:
It is possible that the triggers we receive as props don't have any matching account, in which case we probably want to ignore them.

It happens when we delete an account — the redux store of the Home doesn't know the trigger has been deleted, and will still pass it to the KonnectorModal component. This was a last minute fix, there's probably a better way to handle this.